### ISSUE TITLE
Point sizes not always absolute

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#2894]
## 🐛 Bug

Point sizes are absolute relative to image pixels when `Viewer.camera.zoom` is less than the size of the points .
`size=4` gives points with a diameter of 4 image pixels but when zooming in beyond `zoom=4` this is no longer true, they become smaller than 4 image pixels)

![image](https://user-images.githubusercontent.com/7307488/122306295-1689a080-cf00-11eb-8a0d-5c1fcdc4b03c.png)
In the image above, at zoom 2.4 and zoom 4 the points are exactly 4 image pixels. At zoom 10 they are significantly smaller

## To Reproduce
```python
import napari
import numpy as np

viewer = napari.Viewer()

image = np.zeros((100, 100))
image[::2, ::2] = 1

viewer.add_image(image)

x = np.linspace(4, 94, 10, endpoint=True)
yy, xx = np.meshgrid(x, x)
yx = np.vstack((yy.flatten(), xx.flatten())).transpose()

viewer.add_points(yx, size=4, face_color='cornflowerblue')
```

1. run above example
2. play with zoom
3. note how point diameters are constant 4 image pixels (seen on grid below) at `Viewer.camera.zoom` < 4
4. note how point diameters become smaller than 4 image pixels at `Viewer.camera.zoom` > 4

cc @brisvag

**Comment by brisvag** [comment]
As we already discovered, I cannot reproduce on my work computer. Apparently, I also cannot reproduce on my laptop. This is zoom 32:

![image](https://user-images.githubusercontent.com/23482191/122346630-78222d00-cf49-11eb-8ca2-775fe647eb06.png)

```yaml
napari: 0.4.9
Platform: Linux-5.12.9-arch1-1-x86_64-with-glibc2.33
System: Arch Linux
Python: 3.9.5 (default, May 24 2021, 12:50:35) [GCC 11.1.0]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.20.1
SciPy: 1.6.3
Dask: 2.30.0
VisPy: 0.6.6

OpenGL:
- GL version: 4.6.0 NVIDIA 465.31
- MAX_TEXTURE_SIZE: 32768

Screens:
- screen 1: resolution 1920x1080, scale 1.0

Plugins:
- console: 0.0.3
- mrcfile-reader: 0.1.2
- napari-em-reader: 0.1.0
- napari-properties-plotter: 0.1.0
- scikit-image: 0.4.8
- svg: 0.1.5
```

**Comment by alisterburt** [comment]
really strange, thanks for the comparison @brisvag - here is my environment for comparison

```yaml
napari: 0.4.9
Platform: macOS-11.2.1-x86_64-i386-64bit
System: MacOS 11.2.1
Python: 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:13:15)  [Clang 11.1.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.20.3
SciPy: 1.6.3
Dask: 2021.05.0
VisPy: 0.6.6

OpenGL:
  - GL version:  2.1 Metal - 71.0.7
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1440x900, scale 2.0

Plugins:
  - console: 0.0.3
  - scikit-image: 0.4.9
  - svg: 0.1.5

```

**Comment by jni** [comment]
Based on vispy/vispy#2078, it looks like this is a GL-driver-specific issue and therefore not something napari can do much about, so I think we should close this. @alisterburt feel free to reopen if you think I've made an erroneous assessment!

**Comment by andy-sweet** [comment]
No need to reopen this, but it looks like the visualized diameter of the points is never 4 at any of these zooms. At a zoom of 2.4 and 4, in the original image, it looks like the diameter is about 5.75.

@alisterburt : am I missing something about the setup here, or can I simply not count? As part of my work on #3132, I'm trying to understand if the size we pass to vispy has a direct correspondence to some size in world coordinates. Based on this, it seems like it doesn't.

**Comment by alisterburt** [comment]
you're right @andy-sweet - I just can't count! I think you're right about the size correspondance, maybe we should try to fix that

**Comment by alisterburt** [comment]
I'm reopening this because of what @andy-sweet mentions in his comment above - point sizes should be absolute from the napari side (with the caveat that they only increase up to a limit set by a combination of GL/drivers/hardware)

**Comment by andy-sweet** [comment]
I'll leave it up to you, but I think it might be better to keep this closed for a couple of reasons.

1. The original bug was about inconsistent sizing at different zoom levels, but this seems to be a vispy issue, and there's not much we can to fix it in napari.
2. The [docs on vispy's VisualMarker](https://vispy.org/api/vispy.visuals.markers.html#vispy.visuals.markers.MarkersVisual) tell us that the the size is in pixels, but they don't mention that it will correspond to the diameter of a disc if that symbol is being used. I guess we could add a scaling factor to make that the case, but I don't think it's too important and it doesn't block my work on #3132.

**Comment by alisterburt** [comment]
Cool, I'll leave open because of point 2 - it really feels to me like the diameter of the disc/longest axis of the marker should be absolute, I was imagining a simple scale-factor on the napari size as you suggest. Glad it doesn't block you on #3132 though

**Comment by sofroniewn** [comment]
I completely missed this when it got posted. I havn't had time to read yet, but I'd like too

**Comment by sofroniewn** [comment]
@kevinyamauchi given you're digging into some of the points sizing issues in vispy can you take a look at this one too see https://github.com/napari/napari/issues/2894#issuecomment-864760708 and linked vispy issue for more detail

**Comment by brisvag** [comment]
~This will be addressed by vispy/vispy#2209!~ Still limited by hardware, unfortunately.

**Comment by brisvag** [comment]
Size relative to data was fixed by #3494.

```python
import napari
import numpy as np

viewer = napari.Viewer()

image = np.zeros((100, 100))
image[::2, ::2] = 1

viewer.add_image(image)

x = np.linspace(4, 94, 10, endpoint=True)
yy, xx = np.meshgrid(x, x)
yx = np.vstack((yy.flatten(), xx.flatten())).transpose()

viewer.add_points(yx, size=5, face_color='cornflowerblue', edge_width=0)

napari.run()
```

https://user-images.githubusercontent.com/23482191/144435651-6120e9e9-056e-4cb0-96b7-c1106a971a82.mp4

The original issue (point size having a hard limit on mac, as per vispy/vispy#2078) is unfortunately not solvable here. We can either close this and leave it, or open another issue just for that so people can find it easily by searching the bug.

**Comment by alisterburt** [comment]
Thanks @brisvag - will close this!

**Comment by yfukai** [comment]
I encountered something similar with 3D points: when I changed the scale, there seemed to be a maximum size limit of points. I was wondering if this is an expected behavior, or if it's possibly a bug related to this. 
I used `napari=0.4.17` and `vispy=0.11.0` installed from `conda-forge` (in M1 Mac).

A code to reproduce this:
```python
import napari
viewer = napari.Viewer()
viewer.add_points([[0, 0, 0], [0, 100, 100]],size=[10,100],shading="spherical")
napari.run()
```

https://github.com/napari/napari/assets/5919272/051ee3c7-e832-4e17-8d61-cbb1e49b845a
